### PR TITLE
adding logic to check if Key Vault or SQL server exists.

### PR DESCRIPTION
**Pull request by msalemcode**
Adding logic to deployment script to exit if Key Vault exists and deleted and give information how to proceed.

### CODE REVIEW

**Review by code4clouds — CHANGES_REQUESTED** [review]
This has some of your environment and not the params.

**Review by code4clouds — CHANGES_REQUESTED** [review]

File: deployment/Deploy.ps1 [hunk 1]
```diff
@@ -120,6 +120,41 @@ Write-Host "🔑 Azure Subscription '$AzureSubscriptionID' selected."
...
+$kv_check=az rest --method post --uri $KeyVaultApiUri --headers 'Content-Type=application/json' --body $KeyVaultApiBody | ConvertFrom-Json
+
+if( $kv_check.reason -eq "AlreadyExists")
+{
+	Write-Host ""
+	Write-Host "KeyVault name already exists."
+	Write-Host "To Purge KeyVault please use the following doc:"
+	Write-Host "https://learn.microsoft.com/en-us/cli/azure/keyvault?view=azure-cli-latest#az-keyvault-purge."
+	Write-Host "You could use new KeyVault name by using parameter -KeyVault"
+	Throw "🛑 KeyVault name already exists."
```
Comment (code4clouds) [inline deployment/Deploy.ps1:139]: Do not throw the error, instead provide the message as is and exit with 1

File: deployment/Deploy.ps1 [hunk 2]
```diff
@@ -120,6 +120,41 @@ Write-Host "🔑 Azure Subscription '$AzureSubscriptionID' selected."
...
+#endregion
+
+#region Check If SQL Server Exist
+$sql_exists = Get-AzureRmSqlServer -ServerName $SQLServerName -ResourceGroupName $ResourceGroupForDeployment -ErrorAction SilentlyContinue
+if ($sql_exists) 
+{
+	Write-Host ""
+	Write-Host "SQl Server name already exists."
+	Write-Host "Please delete existing instance or use new sql Instance name by using parameter -SQLServerName"
+	Throw "🛑 SQL Server name already exists."
```
Comment (code4clouds) [inline deployment/Deploy.ps1:153]: Do not throw the error, instead provide the message as is and exit with 1

File: deployment/Deploy.ps1 [hunk 3]
```diff
@@ -120,6 +120,41 @@ Write-Host "🔑 Azure Subscription '$AzureSubscriptionID' selected."
...
+
+$KeyVaultApiUri="https://management.azure.com/subscriptions/$AzureSubscriptionID/providers/Microsoft.KeyVault/checkNameAvailability?api-version=2019-09-01"
+$KeyVaultApiBody='{"name": "'+$KeyVault+'","type": "Microsoft.KeyVault/vaults"}'
+
+$kv_check=az rest --method post --uri $KeyVaultApiUri --headers 'Content-Type=application/json' --body $KeyVaultApiBody | ConvertFrom-Json
+
+if( $kv_check.reason -eq "AlreadyExists")
+{
+	Write-Host ""
+	Write-Host "KeyVault name already exists."
```
Comment (code4clouds) [inline deployment/Deploy.ps1:135]: Consider using the emoji :-) as in:

🛑KeyVault name already exists.

File: deployment/Deploy.ps1 [hunk 4]
```diff
@@ -120,6 +120,41 @@ Write-Host "🔑 Azure Subscription '$AzureSubscriptionID' selected."
...
+
+
+#endregion
+
+#region Check If SQL Server Exist
+$sql_exists = Get-AzureRmSqlServer -ServerName $SQLServerName -ResourceGroupName $ResourceGroupForDeployment -ErrorAction SilentlyContinue
+if ($sql_exists) 
+{
+	Write-Host ""
+	Write-Host "SQl Server name already exists."
```
Comment (code4clouds) [inline deployment/Deploy.ps1:151]: Consider using the emoji :-) as in:

🛑Sql Server name $ServerName already exists.

**Review by code4clouds — CHANGES_REQUESTED** [review]

**Review by code4clouds — APPROVED** [review]
@msalemcode Its all good... just add an extra space on the emoji and merge.

**Review by dstarr — APPROVED** [review]
Works fine with new install and new installs with pre-existing artifacts.

**Review by code4clouds — APPROVED** [review]

**Comment by code4clouds** [comment]
@msalemcode - Code approved, pleases merge.

**Review by code4clouds — APPROVED** [review]

**Comment by santhoshb-msft** [comment]
The PR is still not merged, @msalem have you seen this #568 ? maybe a better approach? I dont if it handles unpurged ones though?

**Pull request merged**